### ISSUE TITLE
[flaky CI] Virtual Machines Check the client When all the VMs are serviced by default (opt-out mode) and trying to create a VM and mac-pool is full should return an error because no MAC address is available

### ISSUE THREAD

**Issue by shwetaap** [kubevirt/cluster-network-addons-operator#572]
**What happened**:
These two tests failed with similar error
[rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:component]Virtual Machines Check the client When all the VMs are serviced by default (opt-out mode) and trying to create a VM after a MAC address has just been released due to a VM deletion and mac-pool is full should re-use the released MAC address for the creation of the new VM and not return an error, using automatic mac allocation of the mac left on the pool

https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/555/pull-e2e-cluster-network-addons-operator-kubemacpool-functests/1306496373650100224

rfe_id:3503][crit:medium][vendor:cnv-qe@redhat.com][level:component]Virtual Machines Check the client When all the VMs are serviced by default (opt-out mode) and trying to create a VM and mac-pool is full should return an error because no MAC address is available
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_cluster-network-addons-operator/565/pull-e2e-cluster-network-addons-operator-kubemacpool-functests/1308392999180308480

This is the error seen

```
/tmp/deploy.kubemacpool.5OGL/github.com/k8snetworkplumbingwg/kubemacpool/tests/virtual_machines_test.go:607
Should succeed parsing the filler vm mac
Unexpected error:
    <*net.AddrError | 0xc0009be900>: {
        Err: "invalid MAC address",
        Addr: "",
    }
    invalid MAC address
occurred
/tmp/deploy.kubemacpool.5OGL/github.com/k8snetworkplumbingwg/kubemacpool/vendor/github.com/onsi/gomega/internal/assertion/assertion.go:43
```
**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:

/kind bug
/triage build-officer

**Comment by RamLavi** [comment]
triage/build-watcher

**Comment by RamLavi** [comment]
/should be fixed when #567 is merged
